### ISSUE TITLE
Train answer quality scorer model to validate extractions

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#567]
## Problem

Extraction accuracy is 2% (1/50 correct) with 43.5% failures due to selecting wrong entities (Issue #554).

Even after fixing extraction logic to consider multiple candidates, we need a learned model to score answer plausibility.

## Solution

Train a small answer quality scorer (~200K params) to validate extracted answers.

**Architecture:**
```python
class AnswerQualityScorer(nn.Module):
    def __init__(self, compositional_embedding):
        self.comp_emb = compositional_embedding  # Frozen 128D
        self.scorer = nn.Sequential(
            nn.Linear(128 * 3, 256),  # question + answer + document
            nn.ReLU(),
            nn.Dropout(0.2),
            nn.Linear(256, 128),
            nn.ReLU(),
            nn.Linear(128, 1),  # quality score
            nn.Sigmoid()
        )
```

**Training data:**
- Positive examples: Correct extractions from evaluation
- Negative examples: Wrong extractions from evaluation
- Bootstrap from current evaluation results

**Integration:**
```python
# In answer_extractor.py
candidates = extract_all_matching_entities(doc)
scored = [(c, quality_scorer(query, c, doc)) for c in candidates]
best = max(scored, key=lambda x: x[1])
```

**Expected impact:** Reduce wrong extraction failures by 50% (from 43.5% to ~20%)

## Dependencies

- Depends on #554 (multi-candidate extraction must be implemented first)
- Depends on #549 (virtual predicates improve extraction coverage)

## Implementation

1. Generate training data from evaluation results
2. Train scorer model (~2 hours)
3. Integrate into answer_extractor.py
4. Re-evaluate on test set

**Estimated time:** 2-3 days
**Params:** ~200K (total system: 320K + 180K + 200K = 700K)

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
